apacheconfig/reader.py

```diff
@@ -12,26 +13,45 @@ def __init__(self):
     def environ(self):
         return self._environ
 
-    def basename(self, filepath):
-        return self._path.basename(filepath)
-
-    def dirname(self, filepath):
-        return self._path.dirname(filepath)
-
     def exists(self, filepath):
         return self._path.exists(filepath)
 
-    def isabs(self, filepath):
-        return self._path.isabs(filepath)
-
     def isdir(self, filepath):
         return self._path.isdir(filepath)
 
-    def join(self, path, *paths):
-        return self._path.join(path, *paths)
-
     def listdir(self, filepath):
         return self._os.listdir(filepath)
 
     def open(self, filename, mode='r', bufsize=-1):
         return open(filename, mode, bufsize)
+
+
+class TestInfraHostReader(object):
```

Comment:
The `TestInfraHostReader` class seems to be very specific to the `TestInfra` project, not so much to `apacheconfig`. Also, it can't be readily used without `host` implementation being provided.

How useful this code would be for a random `apacheconfig` user? If it is not, should we move it to `TestInfra`?

Reply:
I was unsure about this too, I first had it in `TestInfra` but then I thought it might be useful as an example. But if you are planning to maybe provide some more general reader implementations, those would probably serve this purpose better.

Reply:
I think so. I'd better add FTP or .zip reader because those would work out of the box.